pages/training/[id].js

```diff
@@ -14,10 +15,19 @@ export default function CollectionPage({
   const router = useRouter();
   const { id } = router.query;
 
-  const allFlashcardsFromCollection = getAllFlashcardsFromCollection(id);
+  const [sortedFlashcards, setSortedFlashcards] = useState([]);
 
-  const collectionTitle = allFlashcardsFromCollection?.[0]?.collectionTitle;
-  const collectionColor = allFlashcardsFromCollection?.[0]?.collectionColor;
+  useEffect(() => {
+    if (id) {
+      const allFlashcardsFromCollection = getAllFlashcardsFromCollection(id);
+      setSortedFlashcards(sortFlashcardsByLevel(allFlashcardsFromCollection));
+    }
+  }, [id, getAllFlashcardsFromCollection]);
```

Comment:
If I don't put in an useEffect Hook I get an error message saying that there are too many re-renders. React limits the number of renders to prevent an infinite loop. I put in the early return so when there is no id it stops but I left the useEffect in.